Documents/Automation/CreateImage/Documents/aws-CreateImage.json

```diff
@@ -0,0 +1,29 @@
+{
+  "description": "Creates a new Amazon Machine Image (AMI) from an Amazon EC2 instance",
+  "schemaVersion": "0.3",
+  "assumeRole": "{{ AutomationAssumeRole }}",
+  "parameters": {
+    "InstanceId": {
+      "type": "String",
+      "description": "(Required) The ID of the Amazon EC2 instance."
+    },
+    "AutomationAssumeRole": {
+      "type": "String",
+      "description": "(Optional) The ARN of the role that allows Automation to perform the actions on your behalf. ",
+      "default": ""
+    }
+  },
+  "mainSteps": [
+{
+    "name": "createImage",
```

Comment:
https://docs.aws.amazon.com/systems-manager/latest/userguide/automation-actions.html#automation-action-create - In the "Important" section it is mentioned that the instance needs to be stopped before invoking create image. So you might want to consider adding additional steps to check if the instance is stopped, if not stop it and restart it after create image.